portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +26,15 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, looks like you didn't give anything.</h1>");
+    } else if (!name.matches("^[A-Za-z0-9 ]*$")){
```

Comment:
Please create a const static variable for "^[A-Za-z0-9 ]*$".

Reply:
Okay, I made and pushed the changes you suggested.